diag_manager/fms_diag_yaml_object.F90

```diff
@@ -34,31 +34,13 @@ module fms_diag_yaml_object_mod
 use iso_c_binding
       implicit none
 integer, parameter :: NUM_REGION_ARRAY = 8
-  !> @brief The files type matching a C struct containing diag_yaml information
-    !> @ingroup fms_diag_files_mod
-type, bind(c) :: diag_yaml_files_struct
-     character (kind=c_char) :: file_fname (20) !< file name
-     character (kind=c_char) :: file_frequnit (7) !< the frequency unit
-     integer (c_int)    :: file_freq !< the frequency of data
-     character (kind=c_char) :: file_timeunit(7) !< The unit of time
-     character (kind=c_char) :: file_unlimdim(8) !< The name of the unlimited dimension
-     character (kind=c_char) :: file_write (5) !< false if the user doesn’t want the file to be 
-                                               !! created (default is true).
-     character (kind=c_char) :: file_realm (3) !< The modeling realm that the variables come from
-     real (c_float) :: file_region (NUM_REGION_ARRAY) !< Bounds of the regional section to capture
-     integer (c_int) :: file_new_file_freq !< Frequency for closing the existing file
-     character (kind=c_char) :: file_new_file_freq_units (3)!< Time units for creating a new file. 
-                                                        !! Required if “new_file_freq” used
-     integer (c_int) :: file_start_time !< Time to start the file for the first time. Requires “new_file_freq”
-     integer (c_int) :: file_duration !< How long the file should receive data after start time 
-                                      !! in “file_duration_units”.  This optional field can only 
-                                      !! be used if the start_time field is present.  If this field
-                                      !! is absent, then the file duration will be equal to the
-                                      !! frequency for creating new files.  
-                                      !! NOTE: The file_duration_units field must also be present if
-                                      !! this field is present.
-    character (kind=c_char) :: file_duration_units (3)!< The file duration units
-end type diag_yaml_files_struct
+
+type sub_region_type
+     character (len=:), allocatable :: grid_type !< Flag indicating the type of region,
+                                                      !!acceptable values are "latlon" and "index"
+     real :: lat_lon_region (NUM_REGION_ARRAY) !< Bounds of the regional section to capture if in "latlon" mode
+     integer :: index_region (NUM_REGION_ARRAY) !< Bounds of the regional section to capture if in "index" mode
+end type sub_region_type
```

Comment:
For the sub-regions, I want to ensure I understand the intent.  This storage object will accommodate the use of four pairs of lat-lon values or i-j indexes.  What I don't see is some way of constraining or limiting the vertical extent nor specifying the particular tile for the index option.

Reply:
> What I don't see is some way of constraining or limiting the vertical extent

I am not sure I understand what you mean. The `*_region` arrays store the bounds for all four of the possible dimensions: [dim1_min, dim1_max, dim2_min, dim2_max, dim3_min, dim3_max, dim4_min, dim4_max].

> specifying the particular tile for the index option.

Yes, I forgot to add the "tile". It will be a required key if using the "index" approach. This will be added in the next commit.

Reply:
This is where confusion comes in on how a sub-region is being defined.  I know in the past we've always defined the sub-region to have sides of constant lat or lon.  I saw the eight entries here and thought maybe we were expanding the capability to create somewhat irregular sub-regiosn denoted by either (i,j) or (lat,lon) for each fo the corners of the sub-region se, sw, ne, nw.